### DISCUSSION TITLE
Mutlicluster Setup: How to Install Istio CRDs in Remote Kubernetes Cluster

### DISCUSSION THREAD

**Discussion by akefirad** [istio/istio#49348] [Q&A]
To avoid any XY problem, I'm sharing the full story.  
I'd like to deploy a multi-cluster setup of Istio; specifically Primary-Remote (single mesh, single network) setup. Following [this][1] guide after a lot of troubleshooting, finally I managed to deploy both clusters with the required resources and tested and everything working just fine.  
Now the issue is that when I try to deploy a VirtualService to the remote cluster, K8S complains that the it doesn't know VirtualService (for obvious reason that the Istio CRDs are not installed in the remote cluster).  
Now my questions are:
1. Shouldn't we also install the CRDs in the remote cluster? Or am I missing something?
2. If yes, then how can install only CRDs using `istioctl`?

Thanks.

  [1]: https://istio.io/latest/docs/setup/install/multicluster/primary-remote/

**Comment by howardjohn** [comment]
Istiod only reads CRs from one cluster, so there is no need to install the CRDs in the remotecluster; if you did, they would just be ignored

**Reply by akefirad** [reply]
Thanks for the reply. But then how do I manage the communication using VirtualService and/or DestinationRule?
Does that mean if I want to deploy such resources in the remote cluster I cannot use Primary-Remote setup?

**Reply by howardjohn** [reply]
You would deploy them all in the primary cluster. Even without primary-remote you would still need to have policies in _all_ clusters since most policies are enforced by clients so for any cross-cluster communication work properly they would all need the configuration

**Reply by akefirad** [reply]
> Even without primary-remote you would still need to have policies in all clusters since most policies are enforced by clients so for any cross-cluster communication work properly they would all need the configuration.

Sorry, I'm not sure if I understand this. In the Primary-Remote setup I expect the istio-proxy sidecars (i.e. clients?) to obtain the configuration from the control plane in the primary cluster. Is that not correct?

> Istiod only reads CRs from one cluster

I was hoping it's possible to some how customize the solution here to have one control plane (in the primary cluster) and configurations (e.g. VirtualService, DestinationRule) in both clusters. Does your statement means this is not possible?

Thanks.

**Reply by dminca** [reply]
@akefirad that sounds more like [Endpoint discovery with multiple control planes](https://istio.io/latest/docs/ops/deployment/deployment-models/#endpoint-discovery-with-multiple-control-planes) or did I misunderstood
